lib/serverless.js

```diff
@@ -215,6 +220,49 @@ class Serverless {
   logDeprecation(code, message) {
     return this._logDeprecation(`EXT_${ensureString(code)}`, ensureString(message));
   }
+
+  extendConfiguration(configurationPathKeys, value) {
+    ensureArray(configurationPathKeys, {
+      ensureItem: ensureString,
+    });
+    if (configurationPathKeys.length < 1) {
+      throw new ServerlessError(
+        'ConfigurationPathKeys needs to contain at least one element.',
+        'INVALID_EXTEND_AFTER_INIT'
+      );
+    }
+
+    if (!this.isConfigurationExtendable) {
+      throw new ServerlessError(
+        'ExtendConfiguration cannot be used after init.',
+        'INVALID_EXTEND_AFTER_INIT'
+      );
+    }
+    try {
+      value = JSON.parse(JSON.stringify(value));
+    } catch (error) {
+      throw new ServerlessError(
+        'ExtendConfiguration called with invalid data. Value is not json-serializable. Error: ${error}',
+        'INVALID_EXTEND_VALUE'
+      );
+    }
+
+    const configurationPath = configurationPathKeys.join('.');
```

Comment:
Let's not join values. I think `_.set` accepts also array of strings as an argument